In [18]:
import pandas as pd
from sklearn.datasets import fetch_california_housing


###
## Data Analysis
###

In [19]:
california = fetch_california_housing()

a = pd.DataFrame( data = california.data, columns = california.feature_names)
a.head()


MedInc  HouseAge  AveRooms  AveBedrms  Population  AveOccup  Latitude  \
0  8.3252      41.0  6.984127   1.023810       322.0  2.555556     37.88   
1  8.3014      21.0  6.238137   0.971880      2401.0  2.109842     37.86   
2  7.2574      52.0  8.288136   1.073446       496.0  2.802260     37.85   
3  5.6431      52.0  5.817352   1.073059       558.0  2.547945     37.85   
4  3.8462      52.0  6.281853   1.081081       565.0  2.181467     37.85   

   Longitude  
0    -122.23  
1    -122.22  
2    -122.24  
3    -122.25  
4    -122.25

###
## Creating Our DataFrame
###

In [20]:
california = fetch_california_housing()

a = pd.DataFrame( data = california.data, columns = california.feature_names)
a['Price'] = california.target
a.to_csv('california dataset', index = False)
b = a.head(500)
b.to_csv('california500', index = False)
b

MedInc  HouseAge  AveRooms  AveBedrms  Population  AveOccup  Latitude  \
0    8.3252      41.0  6.984127   1.023810       322.0  2.555556     37.88   
1    8.3014      21.0  6.238137   0.971880      2401.0  2.109842     37.86   
2    7.2574      52.0  8.288136   1.073446       496.0  2.802260     37.85   
3    5.6431      52.0  5.817352   1.073059       558.0  2.547945     37.85   
4    3.8462      52.0  6.281853   1.081081       565.0  2.181467     37.85   
..      ...       ...       ...        ...         ...       ...       ...   
495  5.3794      52.0  6.861017   1.123729      1316.0  2.230508     37.86   
496  3.2619      52.0  4.792053   1.017219      1508.0  1.997351     37.85   
497  2.7054      52.0  5.001227   1.128834      1800.0  2.208589     37.85   
498  2.2431      47.0  6.371166   1.226994       719.0  2.205521     37.85   
499  1.9028      50.0  5.015686   1.176471       675.0  2.647059     37.85   

     Longitude  Price  
0      -122.23  4.526  
1      -122.22  3.585  
2      -122.24  3.521  
3      -122.25  3.413  
4      -122.25  3.422  
..         ...    ...  
495    -122.25  3.769  
496    -122.26  3.096  
497    -122.27  1.823  
498    -122.27  1.727  
499    -122.27  1.508  

[500 rows x 9 columns]

###
## Description
###

In [21]:
a.describe()

MedInc      HouseAge      AveRooms     AveBedrms    Population  \
count  20640.000000  20640.000000  20640.000000  20640.000000  20640.000000   
mean       3.870671     28.639486      5.429000      1.096675   1425.476744   
std        1.899822     12.585558      2.474173      0.473911   1132.462122   
min        0.499900      1.000000      0.846154      0.333333      3.000000   
25%        2.563400     18.000000      4.440716      1.006079    787.000000   
50%        3.534800     29.000000      5.229129      1.048780   1166.000000   
75%        4.743250     37.000000      6.052381      1.099526   1725.000000   
max       15.000100     52.000000    141.909091     34.066667  35682.000000   

           AveOccup      Latitude     Longitude         Price  
count  20640.000000  20640.000000  20640.000000  20640.000000  
mean       3.070655     35.631861   -119.569704      2.068558  
std       10.386050      2.135952      2.003532      1.153956  
min        0.692308     32.540000   -124.350000      0.149990  
25%        2.429741     33.930000   -121.800000      1.196000  
50%        2.818116     34.260000   -118.490000      1.797000  
75%        3.282261     37.710000   -118.010000      2.647250  
max     1243.333333     41.950000   -114.310000      5.000010

**Statistical Summary:**

- Count: The number of data points for each attribute (20640 observations).
- Mean: The average value of each attribute.
- Std (Standard Deviation): A measure of the variation or dispersion of each attribute.
- Min: The minimum value of each attribute.
- 25% (First Quartile): The value below which 25% of the data fall.
- 50% (Median/Second Quartile): The value below which 50% of the data fall.
- 75% (Third Quartile): The value below which 75% of the data fall.
- Max: The maximum value of each attribute.

###
## info
###

In [22]:
b.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   MedInc      500 non-null    float64
 1   HouseAge    500 non-null    float64
 2   AveRooms    500 non-null    float64
 3   AveBedrms   500 non-null    float64
 4   Population  500 non-null    float64
 5   AveOccup    500 non-null    float64
 6   Latitude    500 non-null    float64
 7   Longitude   500 non-null    float64
 8   Price       500 non-null    float64
dtypes: float64(9)
memory usage: 35.3 KB


###
## Sum of Null Values In Our DataFrame
###

In [23]:
b.isnull().sum()

MedInc        0
HouseAge      0
AveRooms      0
AveBedrms     0
Population    0
AveOccup      0
Latitude      0
Longitude     0
Price         0
dtype: int64